### PR TITLE
Adds utility functions for code and seqD generation from stringified SD

### PR DESCRIPTION
**Pull request by miguelrk**
This PR introduces a new `index.ts` file under /src with utility functions for typescript code (main and base) and sequence diagram generations from a stringified SD, rather than having to use files. This allows a more natural workflow for API-based projects, where users can simply import these functions and use the code and seqD generation algorithms on demand, from strings.

Additionally, the internally used `transpileTStoJS` is also exported in case it might be useful to transpile TS to JS from a string programatically.

These functions import and use the already available functions from this package internally (e.g. `codeGen`, `parseSD`, etc)

**The 3 exported utility functions are:**

```typescript
export function getSeqD(mashupLogic: SQSD.mashupLogic, systemName: string): string

export async function getCode(
  sd: SQSD.sdTemplate,
  mashupLogic: SQSD.mashupLogic,
  systemName: string,
  outFileName = 'codeTS'
): Promise<{ codeTS: string; baseTS: string }>

/**
 * Transpiles a TypeScript string (with ES module import/export syntax by default)
 * to its valid JavaScript string representation using the native ts.transpileModule method
 *
 * @see https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API#a-simple-transform-function
 *
 * @param codeTS the typescript code to be transpiled
 * @returns a javascript string of the transpiled code
 */
export function transpileTStoJS(codeTS: string, moduleType: moduleType = 'CommonJS'): string
```

### CODE REVIEW

**Review by egekorkan — APPROVED** [review]

**Comment by egekorkan** [comment]
Looks good to me, @adriankast ?

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/* eslint-disable */
+// import "./sdToSeqD";
+import checkSD from './validateSd';
+import { SQSD } from '../definitions/defs';
+import parseSD from './parseSD';
+import generateTS from './codeGen';
+// import generateSD from "./generateSD";
```
Comment (adriankast) [inline src/index.ts:7]: Same as above

**Review by adriankast — APPROVED** [review]
Just asking about comments, looks good for me too 👍

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+/* eslint-disable */
+// import "./sdToSeqD";
```
[the remark this answers is not in the document]
Comment (miguelrk) [reply src/index.ts:2]: `/* eslint-disable */` is indeed required, else we get some unavoidable linting errors if I remember correctly. `// import "./sdToSeqD";` can indeed be removed. Good catch

File: src/index.ts (hunk 1 above)
Comment (miguelrk) [reply src/index.ts:7]: `// import generateSD from "./generateSD"` could also be removed. I left it there explicitly as to show how missing functionality could eventually be implemented in the same way (in this case, SD generation from SeqD). We can safely remove it though..

**Comment by egekorkan** [comment]
I think these comments should be removed. Could you remove them @miguelrk ?

**Comment by miguelrk** [comment]
> I think these comments should be removed. Could you remove them @miguelrk ?

Of course, I just removed them in 87e85e8

**Pull request merged**